Description: Locate libvncserver/client with pkg-config instead of
 libvncserver-config. - The latter causes unnecessary linkage.
Author: Andreas Metzler <ametzler@debian.org>
Author: Helmut Grohne <helmut@subdivi.de>
Last-Update: 2016-11-17

---
 configure.ac |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -17,6 +17,7 @@ AC_PROG_MAKE_SET
 AC_PROG_RANLIB
 AC_PATH_PROG([AR], [ar], [/usr/bin/ar],
                   [$PATH:/usr/ccs/bin])
+PKG_PROG_PKG_CONFIG
 
 # Options
 AH_TEMPLATE(WITH_TIGHTVNC_FILETRANSFER, [Disable TightVNCFileTransfer protocol])
@@ -414,9 +415,9 @@ if test "x$with_avahi" != "xno"; then
 		AVAHI_LIBS="-L$with_avahi/lib -lavahi-common -lavahi-client"
 		echo "using $with_avahi"
 		with_avahi=yes
-	elif pkg-config --atleast-version=0.6.4 avahi-client >/dev/null 2>&1; then
-		AVAHI_CFLAGS=`pkg-config --cflags avahi-client`
-		AVAHI_LIBS=`pkg-config --libs avahi-client`
+	elif $PKG_CONFIG --atleast-version=0.6.4 avahi-client >/dev/null 2>&1; then
+		AVAHI_CFLAGS=`$PKG_CONFIG --cflags avahi-client`
+		AVAHI_LIBS=`$PKG_CONFIG --libs avahi-client`
 		with_avahi=yes
 		echo yes
 	else
@@ -504,8 +505,8 @@ new enough.
 ==========================================================================
 ])
 		else
-			SYSTEM_LIBVNCSERVER_CFLAGS=`libvncserver-config --cflags`
-			SYSTEM_LIBVNCSERVER_LIBS="$rflag"`libvncserver-config --libs`
+			SYSTEM_LIBVNCSERVER_CFLAGS=`$PKG_CONFIG --cflags libvncserver libvncclient`
+			SYSTEM_LIBVNCSERVER_LIBS="$rflag"`$PKG_CONFIG  --libs libvncserver libvncclient`
 			with_system_libvncserver=yes
 			echo yes
 		fi
